frontend/src/Editor/Viewer.jsx

```diff
@@ -152,19 +157,69 @@ class ViewerComponent extends React.Component {
   };
 
   loadApplicationBySlug = (slug) => {
-    appService.getAppBySlug(slug).then((data) => {
-      this.setStateForApp(data);
-      this.setStateForContainer(data);
-      this.setWindowTitle(data.name);
-      this.setState({ isLoading: false });
-    });
+    appService
+      .getAppBySlug(slug)
+      .then((data) => {
+        this.setStateForApp(data);
+        this.setStateForContainer(data);
+        this.setWindowTitle(data.name);
+      })
+      .catch((error) => {
+        this.setState({
+          errorDetails: error,
+          errorAppId: slug,
+          errorVersionId: null,
+          isLoading: false,
+        });
+      });
   };
 
   loadApplicationByVersion = (appId, versionId) => {
-    appService.getAppByVersion(appId, versionId).then((data) => {
-      this.setStateForApp(data);
-      this.setStateForContainer(data);
-    });
+    appService
+      .getAppByVersion(appId, versionId)
+      .then((data) => {
+        this.setStateForApp(data);
+        this.setStateForContainer(data);
+      })
+      .catch((error) => {
+        this.setState({
+          errorDetails: error,
+          errorAppId: appId,
+          errorVersionId: versionId,
+          isLoading: false,
+        });
+      });
+  };
+
+  switchOrganization = (orgId, appId, versionId) => {
+    const path = `/applications/${appId}${versionId ? `/versions/${versionId}` : ''}`;
+    organizationService.switchOrganization(orgId).then(
+      (data) => {
+        authenticationService.updateCurrentUserDetails(data);
+        window.location.href = path;
+      },
+      () => {
+        return (window.location.href = `login/${orgId}?redirectTo=${path}`);
+      }
+    );
+  };
+
+  handleError = (errorDetails, appId, versionId) => {
+    if (errorDetails?.data) {
+      const statusCode = errorDetails.data?.statusCode;
+      if (statusCode === 403) {
+        const errorObj = JSON.parse(errorDetails.data?.message);
```

Comment:
we should consider the scenario when `errorDetails.data?.message` empty or not valid json. try catch can be added and on error redirected to home page, same for editor loader